pkg/controllers/management/template/template_project_catalog.go

```diff
@@ -0,0 +1,172 @@
+		},
+		{
+			APIGroups:     []string{"management.cattle.io"},
+			Resources:     []string{"templateversions"},
+			ResourceNames: projectTemplateVersions,
+			Verbs:         []string{"get", "list", "watch"},
+		},
+	}
+
+	// Add these templates/templateversions to the clusterRole for this project's templates/templateVersions
+	// check if the clusterRole exists
+	cRole, err := tm.clusterRoleClient.Get(crName, metav1.GetOptions{})
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		// error is IsNotFound, so create it
+		newCR := &v1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crName,
+			},
+			Rules: newRules,
+		}
+		_, err := tm.clusterRoleClient.Create(newCR)
+		if err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	} else {
+		// update the role with the latest templates and templateversions
+		if !reflect.DeepEqual(cRole.Rules, newRules) {
+			// update the cRole
+			cRole = cRole.DeepCopy()
+			cRole.Rules = newRules
+			_, err := tm.clusterRoleClient.Update(cRole)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Now that the cluster role is created/updated, create/update the cluster role binding with appropriate subjects
+	// Get all subjects for this CRB, which includes all users in this projects
+	// get a list of all members/owners of this project using PRTBs
+	prtbs, err := tm.prtbLister.List(projectID, labels.NewSelector())
+	if err != nil {
+		return err
+	}
+	subjects := []v1.Subject{}
+	subjectMap := make(map[v1.Subject]bool)
+	for _, prtb := range prtbs {
+		s, valid, err := buildSubjectFromRTB(prtb)
+		if err != nil {
+			return err
+		}
+		if !valid {
+			continue
+		}
+		if !subjectMap[s] {
+			subjectMap[s] = true
+		}
+	}
+
+	for key := range subjectMap {
+		subjects = append(subjects, key)
+	}
+
+	sort.Slice(subjects, func(i, j int) bool { return subjects[i].Name < subjects[j].Name })
+
+	// Now create or update the  CRB
+	// First check if a clusterRoleBinding exists for the templates/templateversions in this cluster
+	crb, err := tm.crbClient.Get(crbName, metav1.GetOptions{})
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		// crb not found, so create it
+		newCRB := &v1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crbName,
+			},
+			RoleRef: v1.RoleRef{
+				Name: crName,
+				Kind: "ClusterRole",
+			},
+			Subjects: subjects,
+		}
+		_, err := tm.crbClient.Create(newCRB)
+		if err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+		return nil
+	}
+	// crb exists, so update it with the new subjects
+	if !reflect.DeepEqual(crb.Subjects, subjects) {
+		crb = crb.DeepCopy()
+		crb.Subjects = subjects
+		_, err := tm.crbClient.Update(crb)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func roleValidForProjectCatalogTemplates(roleTemplateName string) bool {
+	if rolesWithAccessToProjectCatalogTemplates[roleTemplateName] {
```

Comment:
looks like you can just `return rolesWithAccessToProjectCatalogTemplates[roleTemplateName]`